### PR TITLE
Two Factor: Add wpcom_vip_use_custom_sso filter

### PR DESCRIPTION
**Pull request by justnorris**
This will make `wpcom_vip_should_force_two_factor` more flexible by allowing other third party SSO solutions or custom logic to help figure out whether or not 2FA should be forced.

## Description
This PR will allow customization of whether or not 2FA should be forced. 

Two Usage examples:
* Implement a custom SSO Plugin other than OneLogin or SimpleSaml
* Force 2FA for users that aren't using SSO. For example:

```php
add_filter( 'wpcom_vip_use_custom_sso', function( $enabled ) {
	
	// The user `justnorris` isn't using OneLogin, force 2FA for him
	$user = wp_get_current_user();
	if ( isset( $user->user_login ) && 'justnorris' === $user->user_login  ) {
		return false;
	}

	// Check if OneLogin SSO is enabled
	if ( function_exists( 'is_saml_enabled' ) && is_saml_enabled() ) {
		return true;
	}

	return $enabled;

} );
```

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. Apply PR
2. Validate that nothing has changed to the login functionality when SSO is used
3. Use the code snippet from above to force Two-Factor on your user (You'll likely also need to disable proxy/local checks in `wpcom_vip_should_force_two_factor() )

### CODE REVIEW

File: two-factor.php [hunk 1]
```diff
@@ -47,34 +47,32 @@ function wpcom_vip_should_force_two_factor() {
...
-	if ( function_exists( '\HumanMade\SimpleSaml\instance' ) && \HumanMade\SimpleSaml\instance() ) {
+	// Allow custom SSO solutions
+	if ( wpcom_vip_use_custom_sso() ) {
 		return false;
 	}
 
 	return true;
 }
 
-function wpcom_vip_is_jetpack_authorize_request() {
```
Comment (joshbetz) [inline two-factor.php]: Accidentally deleted this function?

Comment (justnorris) [reply two-factor.php]: 🤔

Comment (justnorris) [reply two-factor.php]: I have no idea why it's red. That function is still there.

Comment (justnorris) [reply two-factor.php]: https://github.com/Automattic/vip-go-mu-plugins/blob/5a5059e1a110a7d41dcecbe4280afe7e377e9f29/two-factor.php#L83-L90

Comment (joshbetz) [reply two-factor.php]: I don't think so https://github.com/Automattic/vip-go-mu-plugins/blob/5a5059e1a110a7d41dcecbe4280afe7e377e9f29/two-factor.php

Comment (joshbetz) [reply two-factor.php]: @justnorris that's a different function. `wpcom_vip_is_jetpack_authorize_request` vs `wpcom_vip_is_jetpack_authorize_rest_request`.

Comment (justnorris) [reply two-factor.php]: Fixed

**Comment by joshbetz** [comment]
r696-vip-go-stacks

**Pull request merged**